### PR TITLE
Feat: Set specific SMTP username

### PR DESCRIPTION
**Pull request by utsavdotpro**
## Summary of change

Accept another field in `smtpSettings` called `username` and use that as the `auth.user` for `nodemailer` transport.

## Related issues
- https://github.com/supertokens/supertokens-node/issues/375

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes
- https://github.com/supertokens/docs/pull/444

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

## Remaining TODOs for this PR
NA

### CODE REVIEW

**Review by rishabhpoddar — APPROVED** [review]

**Comment by rishabhpoddar** [comment]
@utsavdotpro the code change looks good to me. A few points:

Can you please run `npm run build-pretty` and then do a git commit and push?

File: lib/ts/ingredients/emaildelivery/services/smtp.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export interface SMTPServiceConfig {
     };
     port: number;
     secure?: boolean;
+    username?: string;
```
Comment (rishabhpoddar) [inline lib/ts/ingredients/emaildelivery/services/smtp.ts:25]: Do you think we should rename this variable to `authUsername`? To make it clear that it's for auth and not to be confused with `from.name`?

**Review by rishabhpoddar — APPROVED** [review]

**Comment by rishabhpoddar** [comment]
thanks @utsavdotpro we will take it from here :)

File: lib/ts/ingredients/emaildelivery/services/smtp.ts (hunk 1 above)
Comment (utsavdotpro) [reply lib/ts/ingredients/emaildelivery/services/smtp.ts:25]: I am in the favour of keeping it as `username` and I see that you have already merged the PR with `username` 👍

**Comment by utsavdotpro** [comment]
Thanks @rishabhpoddar, waiting for the version update to use it in my project 😄

**Comment by rishabhpoddar** [comment]
Actually, I merged it and changed it to `authUsername`

**Comment by utsavdotpro** [comment]
Oh, I think `username` is better as
- we are using `password` and not `authPassword` for auth.pass so this follows the same pattern
- also, it can't be confused with `from.name` as that is just the name of the sender (hence from.name) as used by all mailing clients

**Pull request merged**